sdk/auth/client.go

```diff
@@ -17,6 +17,26 @@ import (
 	"github.com/hashicorp/go-retryablehttp"
 )
 
+var (
+	// Client is the HTTP client used for sending authentication requests and obtaining tokens
+	Client HTTPClient
+
+	// MetadataClient is the HTTP client used for obtaining tokens from the Instance Metadata Service
+	MetadataClient HTTPClient
```

Comment:
do these need to be public?

Reply:
They do if we want to mock them out I think? But I think this will also address other potential needs for customizing the client(s), especially if you can shim them.

Reply:
I guess if we need them outside of this package, yeah fair point 👍